### PR TITLE
[settings] ensure year string is at the end in clean datetime regexp

### PR DESCRIPTION
**Pull request by mkortstiege**
Fixes scraping issue reported in http://trac.kodi.tv/ticket/16170. Credits @Paxxi

### CODE REVIEW

**Comment by mkortstiege** [comment]
@Montellese and @Paxxi for review/comments please.

**Comment by Paxxi** [comment]
+1

**Comment by mkortstiege** [comment]
@MilhouseVH, could you please integrate this into your rpi builds in order to get some more testers on it before pushing to master? Thanks.

**Comment by MilhouseVH** [comment]
Sure, next build.

**Comment by mkortstiege** [comment]
> Sure, next build.

Great, thanks.

File: xbmc/settings/AdvancedSettings.cpp [hunk 1]
```diff
@@ -208,7 +208,7 @@ void CAdvancedSettings::Initialize()
   m_fullScreenOnMovieStart = true;
   m_cachePath = "special://temp/";
 
-  m_videoCleanDateTimeRegExp = "(.*[^ _\\,\\.\\(\\)\\[\\]\\-])[ _\\.\\(\\)\\[\\]\\-]+(19[0-9][0-9]|20[0-1][0-9])([ _\\,\\.\\(\\)\\[\\]\\-]|[^0-9]$)";
+  m_videoCleanDateTimeRegExp = "(.*[^ _\\,\\.\\(\\)\\[\\]\\-])[ _\\.\\(\\)\\[\\]\\-]+(19[0-9][0-9]|20[0-1][0-9])([ _\\,\\.\\(\\)\\[\\]\\-]|[^0-9]$)?";
```
Comment (MilhouseVH) [inline xbmc/settings/AdvancedSettings.cpp]: It's another 5 years before it becomes an issue, but why do we only match on the first two decades of this century `20[0-1][0-9]` - this seems unnecessarily restrictive and something that will eventually become a problem.

Comment (mkortstiege) [inline xbmc/settings/AdvancedSettings.cpp]: Right, will adjust it later today.

**Comment by mkortstiege** [comment]
Updated.

**Comment by mkortstiege** [comment]
Any objections? Would like to shove this one in.

**Comment by MilhouseVH** [comment]
Not had any complaints from testers so far.

**Pull request merged**